### PR TITLE
[pulpcore] add plugin for pulp-3

### PR DESCRIPTION
**Pull request by pmoravec**
Pulp-3 / pulpcore as a revolution from pulp-2 needs a separate
plugin, since both plugins have nothing in common and there might
be deployments where is active both pulp-2 and pulp-3.

Resolves: #2278

Signed-off-by: Pavel Moravec <pmoravec@redhat.com>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [X] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [X] Is the subject and message clear and concise?
- [X] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [X] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [X] If this commit closes an existing issue, is the line `Closes: #ISSUENUMBER` included in an independent line?
- [X] If this commit resolves an existing pull request, is the line `Resolves: #PRNUMBER` included in an independent line?

### CODE REVIEW

**Comment by pmoravec** [comment]
This is so far a WIP version of the plugin, that needs some review from pulp devels and some more testing.

@goosemania , could you please (ask for a) review of the plugin?

For testing on a sos-3.9 version, it is sufficient to:
- copy the file to `sos/plugins/pulpcore.py`
- `sed 's/sos.report.plugins/sos.plugins/1'`
- `sed 's/IndependentPlugin/RedHatPlugin, DebianPlugin, UbuntuPlugin/g'

File: sos/report/plugins/pulpcore.py
Comment (TurboTurtle) [inline sos/report/plugins/pulpcore.py:1]: Hmm. ;-)

File: sos/report/plugins/pulpcore.py
Comment (TurboTurtle) [inline sos/report/plugins/pulpcore.py:18]: Needs a `short_desc` class attr

**Review by TurboTurtle — COMMENTED** [review]
Initial comments below, very minor points. Only other question I have is why `pulpcore` instead of `pulp3`? I don't necessarily mind using the former, I'm just curious as to the reasoning.

File: sos/report/plugins/pulpcore.py
Comment (pmoravec) [reply sos/report/plugins/pulpcore.py:87]: Yes and no: `curl`-ing localhost fails on SSL certificate/hostname check. But `--insecure` option skips that check, so I will use it.

Technically, the very proper solution is to use `CONTENT_HOST = "my.fqdn"` from `/etc/pulp/settings.py` but 1) we would have to have a failover value and 2) the hostname will very rarely differ from the local host. So till `--insecure` works, I am in favour of `localhost`, for the sake of simplicity.

**Comment by pmoravec** [comment]
> Initial comments below, very minor points. Only other question I have is why `pulpcore` instead of `pulp3`? I don't necessarily mind using the former, I'm just curious as to the reasoning.

No strong argument or opinion here. `pulp-3` is the "branding" name, `pulpcore` is the classes name. Maybe `pulp3` is more informative to users of sosreport, to stress it is the "replacement" of pulp plugin / pulp-2?

IMHO short description should have both terms, plugin name: no much preference (now I tend to `pulp3` bit). If somebody agrees with `pulp3`, let comment it and I will rename the plugin.

**Comment by evgeni** [comment]
We've been using "pulpcore" in various places (Puppet module, Repository name, etc), as we didn't want to hardcode the *3* to differentiate from pulp(2) which is usually referred to as "Pulp".

**Review by TurboTurtle — APPROVED** [review]

**Comment by TurboTurtle** [comment]
Merged via 808d9f3

**Comment by ekohl** [comment]
I missed this PR. Is there a good way to stay on top of these (Foreman, Pulp and the subsystems it depends on)?

**Comment by pmoravec** [comment]
> I missed this PR. Is there a good way to stay on top of these (Foreman, Pulp and the subsystems it depends on)?

Yeah I thought we get some feedback before merging the PR, but never mind, we can iterate / fine-tune the plugin further, for sure. Let open an issue if you spot something (and we can make the filename a variable as well, as a side-fix).

**Comment by ekohl** [comment]
Next time, feel free to ping me in a PR.

**Pull request closed without merging**
